.github/workflows/codeql-analysis.yml

```diff
@@ -11,7 +11,7 @@ on: ["push", "pull_request"]
 jobs:
   analyze:
     name: Analyze
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-24.04
```

Comment:
I wonder if we should convert everything over to `ubuntu-latest` so we don't have to worry about chasing Ubuntu versions?

Reply:
there are basically two options
- pin os version ... thus chasing them (while controlling upgrades)
- use latest os ... hunting portability issues between os version upon upgrade (happening uncontrolled) :-(

nb: such new issues arose with ubuntu 24.04 (while i.e. 22.04 was straight)

I tend towards the first option while automating the check for version upgrades using dependabot :+1:

Reply:
> there are basically two options
> 
> * pin os version ... thus chasing them (while controlling upgrades)
> * use latest os ... hunting portability issues between os version upon upgrade (happening uncontrolled) :-(
> 
> nb: such new issues arose with ubuntu 24.04 (while i.e. 22.04 was straight)
> 
> I tend towards the first option while try to automating the check for version upgrades using dependabot 👍

I also have a slight preference toward the first option.  While not directly applicable to libseccomp, Ubuntu 24.04 now defaults to cgroup v2.  That's a big change for other projects, and it isn't one I would want to automatically pick up without thoroughly thinking through the ramifications.

Reply:
well let's go with the pinned os-version then :+1: